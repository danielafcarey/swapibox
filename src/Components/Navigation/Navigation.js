import React from 'react';
import PropTypes from 'prop-types';

const Navigation = (props) => {
  const { 
    selectedButton, 
    changeCategory, 
    favoritesLength 
  } = props;

  let showSelectMessage;
  selectedButton ? 
    showSelectMessage = 'hide' : 
    showSelectMessage = 'showSelectMessage';

  let peopleClass;
  if (selectedButton === 'People') {
    peopleClass = 'selected'
  }

  let planetsClass;
  if (selectedButton === 'Planets') {
    planetsClass = 'selected'
  }

  let vehiclesClass;
  if (selectedButton === 'Vehicles') {
    vehiclesClass = 'selected'
  }

  let favoritesClass;
  if (selectedButton === 'Favorites') {
    favoritesClass = 'selected'
  }

  return (
    <div className='Navigation'>
      <button 
        className={ peopleClass }
        name='People'
        onClick={ changeCategory } >
        People
      </button>
      <button 
        className={ planetsClass }
        name='Planets'
        onClick={ changeCategory } >
        Planets
      </button>
      <button 
        className={ vehiclesClass }
        name='Vehicles'
        onClick={ changeCategory } >
        Vehicles
      </button>
      <button 
        className={ favoritesClass }
        name='Favorites'
        onClick={ changeCategory } >
        Favorites ({ favoritesLength })
      </button>
      <h3 className={ showSelectMessage }>Select a category</h3>
    </div>
  );
}

Navigation.propTypes = {
  selectedButton: PropTypes.string.isRequired,
  changeCategory: PropTypes.func.isRequired,
  favoritesLength: PropTypes.number
};

export default Navigation;
