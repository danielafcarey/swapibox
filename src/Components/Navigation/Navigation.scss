.Navigation {
  display: inline-block;
  float: right;
  margin: 30px 15px;

  button {
    background: black;
    border: none;
    color: white;
    font-size: 1.6rem;
    margin: 5px;
    font-family: 'Montserrat', sans-serif;

    &:hover {
      cursor: pointer;
    }

    &:focus {
      outline: none;
    }

  }
    button.selected {
      color: #feda4a;
    }

  .showSelectMessage {
    display: block;
    font-size: 2rem;
    text-align: right;
    margin-right: 1rem;
    margin-top: .5rem;
  }

  .hide {
    display: none;
  }
}
