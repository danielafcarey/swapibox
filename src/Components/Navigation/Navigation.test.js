import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

describe('Navigation', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      changeCategory: jest.fn(),
      selectedButton: ''
    };
    wrapper = shallow(<Navigation { ...props } />) 
  })

  it('should match the snapshot', () => {
    props.selectedButton = 'People';

    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if no button is selected', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should change the value of peopleClass if the selectedButton is People', () => {
    props.selectedButton = 'People';
    wrapper = shallow(<Navigation { ...props } />)
    const selectedButton = wrapper.find('button.selected');

    expect(selectedButton.props().name).toEqual('People');
  })

  it('should change the value of planetsClass if the selectedButton is Planets', () => {
    props.selectedButton = 'Planets';
    wrapper = shallow(<Navigation { ...props } />)
    const selectedButton = wrapper.find('button.selected');

    expect(selectedButton.props().name).toEqual('Planets');
  })

  it('should change the value of vehiclesClass if the selectedButton is Vehicles', () => {
    props.selectedButton = 'Vehicles';
    wrapper = shallow(<Navigation { ...props } />)
    const selectedButton = wrapper.find('button.selected');

    expect(selectedButton.props().name).toEqual('Vehicles');
  })

  it('should change the value of favoritesClass if the selectedButton is Favorites', () => {
    props.selectedButton = 'Favorites';
    wrapper = shallow(<Navigation { ...props } />)
    const selectedButton = wrapper.find('button.selected');

    expect(selectedButton.props().name).toEqual('Favorites');
  })



})

