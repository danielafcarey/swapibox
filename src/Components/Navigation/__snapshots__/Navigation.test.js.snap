// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    changeCategory={[Function]}
    selectedButton=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className={undefined}
          name="People"
          onClick={[Function]}
>
          People
</button>,
        <button
          className={undefined}
          name="Planets"
          onClick={[Function]}
>
          Planets
</button>,
        <button
          className={undefined}
          name="Vehicles"
          onClick={[Function]}
>
          Vehicles
</button>,
        <button
          className={undefined}
          name="Favorites"
          onClick={[Function]}
>
          Favorites (
          )
</button>,
        <h3
          className="showSelectMessage"
>
          Select a category
</h3>,
      ],
      "className": "Navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": undefined,
          "name": "People",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": undefined,
          "name": "Planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": undefined,
          "name": "Vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites (",
            undefined,
            ")",
          ],
          "className": undefined,
          "name": "Favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites (",
          undefined,
          ")",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a category",
          "className": "showSelectMessage",
        },
        "ref": null,
        "rendered": "Select a category",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className={undefined}
            name="People"
            onClick={[Function]}
>
            People
</button>,
          <button
            className={undefined}
            name="Planets"
            onClick={[Function]}
>
            Planets
</button>,
          <button
            className={undefined}
            name="Vehicles"
            onClick={[Function]}
>
            Vehicles
</button>,
          <button
            className={undefined}
            name="Favorites"
            onClick={[Function]}
>
            Favorites (
            )
</button>,
          <h3
            className="showSelectMessage"
>
            Select a category
</h3>,
        ],
        "className": "Navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": undefined,
            "name": "People",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": undefined,
            "name": "Planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": undefined,
            "name": "Vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites (",
              undefined,
              ")",
            ],
            "className": undefined,
            "name": "Favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites (",
            undefined,
            ")",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a category",
            "className": "showSelectMessage",
          },
          "ref": null,
          "rendered": "Select a category",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navigation should match the snapshot if no button is selected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    changeCategory={[Function]}
    selectedButton=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className={undefined}
          name="People"
          onClick={[Function]}
>
          People
</button>,
        <button
          className={undefined}
          name="Planets"
          onClick={[Function]}
>
          Planets
</button>,
        <button
          className={undefined}
          name="Vehicles"
          onClick={[Function]}
>
          Vehicles
</button>,
        <button
          className={undefined}
          name="Favorites"
          onClick={[Function]}
>
          Favorites (
          )
</button>,
        <h3
          className="showSelectMessage"
>
          Select a category
</h3>,
      ],
      "className": "Navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": undefined,
          "name": "People",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": undefined,
          "name": "Planets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": undefined,
          "name": "Vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites (",
            undefined,
            ")",
          ],
          "className": undefined,
          "name": "Favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites (",
          undefined,
          ")",
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a category",
          "className": "showSelectMessage",
        },
        "ref": null,
        "rendered": "Select a category",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className={undefined}
            name="People"
            onClick={[Function]}
>
            People
</button>,
          <button
            className={undefined}
            name="Planets"
            onClick={[Function]}
>
            Planets
</button>,
          <button
            className={undefined}
            name="Vehicles"
            onClick={[Function]}
>
            Vehicles
</button>,
          <button
            className={undefined}
            name="Favorites"
            onClick={[Function]}
>
            Favorites (
            )
</button>,
          <h3
            className="showSelectMessage"
>
            Select a category
</h3>,
        ],
        "className": "Navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": undefined,
            "name": "People",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": undefined,
            "name": "Planets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": undefined,
            "name": "Vehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites (",
              undefined,
              ")",
            ],
            "className": undefined,
            "name": "Favorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites (",
            undefined,
            ")",
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a category",
            "className": "showSelectMessage",
          },
          "ref": null,
          "rendered": "Select a category",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
